test/cdp-proxy/reactNativeCDPProxy.test.ts

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import {
+    Connection,
+    Server,
+    WebSocketTransport,
+    IProtocolCommand,
+    IProtocolError,
+    IProtocolSuccess
+} from "vscode-cdp-proxy";
+import {ReactNativeCDPProxy} from "./../../src/cdp-proxy/reactNativeCDPProxy";
+import {generateRandomPortNumber} from "./../../src/common/extensionHelper";
+import {RnCDPMessageHandler} from "../../src/cdp-proxy/CDPMessageHandlers/rnCDPMessageHandler";
+import {ICDPMessageHandler} from "../../src/cdp-proxy/CDPMessageHandlers/ICDPMessageHandler";
+import {DirectCDPMessageHandler} from "../../src/cdp-proxy/CDPMessageHandlers/directCDPMessageHandler";
+import {LogLevel} from "../../src/extension/log/LogHelper";
+import {DebuggerEndpointHelper} from "./../../src/cdp-proxy/debuggerEndpointHelper";
+import {Request} from "../../src/common/node/request";
+import * as assert from "assert";
+import {CDP_API_NAMES} from "../../src/cdp-proxy/CDPMessageHandlers/CDPAPINames";
+import {PromiseUtil} from "../../src/common/node/promise";
+
+suite("reactNativeCDPProxy", function () {
+    const promiseUtil = new PromiseUtil();
+
+    const cdpProxyHostAddress = "127.0.0.1"; // localhost
+    const cdpProxyPort = generateRandomPortNumber();
+    const cdpProxyLogLevel = LogLevel.Custom;
+    const proxy = new ReactNativeCDPProxy(cdpProxyHostAddress, cdpProxyPort);
+
+    const wsTargetPort = generateRandomPortNumber();
+    let wsTargetServer: Server | null;
+
+    let targetConnection: Connection | null;
+    let debugConnection: Connection | null;
+
+    suiteSetup(async () => {
+      proxy.setApplicationTargetPort(wsTargetPort);
+      await proxy.initializeServer(new RnCDPMessageHandler(), cdpProxyLogLevel);
+
+      await Server.create({ host: "localhost", port: wsTargetPort })
+        .then((server: Server) => {
+          wsTargetServer = server;
+
+          server.onConnection(([connection, request]: [Connection, Request]) => {
+            targetConnection = connection;
+          });
+        });
+
+      const proxyUri = await new DebuggerEndpointHelper().getWSEndpoint(`http://${cdpProxyHostAddress}:${cdpProxyPort}`);
+      debugConnection = new Connection(await WebSocketTransport.create(proxyUri));
+
+      while (!targetConnection) {
+        await promiseUtil.delay(1000);
```

Comment:
It seems that we have an infinite loop here, we should explicitly [set a timeout for this](https://mochajs.org/#timeouts)

Reply:
I will leave comments in the code